### PR TITLE
System.ArraySegment F# snippets

### PR DESCRIPTION
**Pull request by albert-du**
Adds F# code snippets for System.ArraySegment<T>

### CODE REVIEW

**Comment by albert-du** [comment]
@dsyme @baronfel Could you review?

File: xml/System/ArraySegment`1.xml [hunk 1]
```diff
@@ -111,9 +112,10 @@
   
 -   Have the same number of elements.  
   
- If you want to retrieve an element by its index in the <xref:System.ArraySegment%601> object, you must cast it to an <xref:System.Collections.Generic.IList%601> object and retrieve it or modify it by using the <xref:System.Collections.Generic.IList%601.Item%2A?displayProperty=nameWithType> property. The following example retrieves the element in an <xref:System.ArraySegment%601> object that delimits a section of a string array.  
+ If you want to retrieve an element by its index in the <xref:System.ArraySegment%601> object, you must cast it to an <xref:System.Collections.Generic.IList%601> object and retrieve it or modify it by using the <xref:System.Collections.Generic.IList%601.Item%2A?displayProperty=nameWithType> property. Note that this is not necessary in F#. The following example retrieves the element in an <xref:System.ArraySegment%601> object that delimits a section of a string array.
```
Comment (baronfel) [inline xml/System/ArraySegment`1.xml:115]: Today I learned a new fact. Nice find and clarification @albert-du 👍

**Review by baronfel — APPROVED** [review]
Examples look good to me, thanks for your persistence on these!

**Review by BillWagner — APPROVED** [review]
Thanks again @albert-du 

I'll :shipit: now.

**Pull request merged**